src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -37,8 +37,7 @@ import java.util.*;
 
 }
 
-LEADING_ASTERISK : ( (' '|'\t') {_tokenStartCharPositionInLine == 0}? ) (' '|'\t')* '*'
-      | '*' {_tokenStartCharPositionInLine == 0}?
+LEADING_ASTERISK : (' '|'\t')* '*'
```

Comment:
JAVADOC -> * Test case for detecting CR newlines.\r* @author Author Name\r<EOF> [0:0]
|--LEADING_ASTERISK -> * [0:0]
|--TEXT ->  Test case for detecting CR newlines. [0:1]
|--NEWLINE -> \r [0:38]
|--LEADING_ASTERISK -> * [0:39]
|--WS ->   [0:40]
|--JAVADOC_TAG -> @author Author Name\r [0:41]
|   |--AUTHOR_LITERAL -> @author [0:41]
|   |--WS ->   [0:48]
|   `--DESCRIPTION -> Author Name\r [0:49]
|       |--TEXT -> Author Name [0:49]
|       `--NEWLINE -> \r [0:60]
`--EOF -> <EOF> [0:61]

Here is the tree, and yes it fails to recognize new lines !